app/lib/tenejo/csv_importer.rb

```diff
@@ -83,57 +83,38 @@ def job_owner
       @depositor
     end
 
-    def search(item, child_id)
-      found = item.children.find { |x| x.identifier == child_id }
-      if !found && !item.children.empty?
-        item.children.each do |x|
-          found ||= search(x, child_id)
-        end
-      end
-      found
+    def search_members(type, file_id)
+      @job.graph.flatten.find { |x| x.is_a?(type) && x.identifier == file_id }
     end
 
-    # TODO: make pfffiles live in the children array, so we don't have to search
-    # for them separately
-    def search_file(item, file_id)
-      # search under the files key instead of children
-      found = item.children.find { |x| x.identifier == file_id } if item.respond_to?(:files)
-      if !found && !item.children.empty?
-        item.children.each do |x|
-          found ||= search_file(x, file_id)
-        end
+    def update_status(item, start_state, end_state)
+      tmp = @job.graph
+      member = search_members(item.class, item.identifier)
+      if member
+        member.status = start_state
+        @job.graph = tmp # this only exists to trick ActiveRecord into actually saving the modified graph
+        @job.save!
+        yield
+        member.status = end_state
+        @job.graph = tmp # same here
+        @job.save!
+      else
+        yield # still need to call the block if the member doesn't exist in the graph somehow. this seems like a bug related to test setup
       end
-      found
-    end
-
-    # TODO: make pfffiles live in the children array, so we don't have to search
-    # for them separately
-    def update_file(file_id, status)
-      file = search_file(@job.graph.root, file_id)
-      return unless file
-      file.status = status
-      @job.save!
-    end
-
-    def update_child(child_id, status)
-      child = search(@job.graph.root, child_id)
-      return unless child
-      child.status = status
-      @job.save!
     end
 
     def create_or_update_collection(pfcollection)
       # put all the expensive stuff here
       # and unit test the heck out of it
-      update_child(pfcollection.identifier, 'started')
-      collection = find_or_new_collection(pfcollection.identifier, pfcollection.title)
-      update_collection_attributes(collection, pfcollection)
-      if pfcollection.parent
-        parent = Collection.where(primary_identifier_ssi: pfcollection.parent).first
-        collection.member_of_collections << parent if parent
+      update_status(pfcollection, 'started', 'completed') do
```

Comment:
The view helper doesn't understand 'started', this needs to be 'in progress' or we need to update the helper:
 
https://github.com/curationexperts/tenejo/blob/v2.19.0/app/helpers/application_helper.rb#L30-L48